Sources/MessagingInApp/Views/InAppMessageView.swift

```diff
@@ -94,6 +106,11 @@ public class InAppMessageView: UIView {
             return // no messages to display, exit early. In the future we will dismiss the View.
         }
 
+        // Do not re-show the existing message if already shown to prevent the UI from flickering as it loads the same message again.
+        if let currentlyShownMessage = inlineMessageManager?.currentMessage, currentlyShownMessage.messageId == messageToDisplay.messageId {
```

Comment:
`currentlyShownMessage.messageId == messageToDisplay.messageId` isn't this template id? so we can have two messages with same template ids and it won;t show the second message because its template id is same as previous one?

Reply:
You're correct, it should be `queueId` and not `messageId`. 

I am pushing up a commit with this change. Unfortunately, we don't have a good way to write a test for this line in this PR. 

[I am adding this case to another ticket](https://linear.app/customerio/issue/MBL-313/after-close-action-pressed-replace-in-app-message-with-next-in-queue) where I'll include an automated test there.